addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlBridgeHandler.java

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.nikohomecontrol.handler;
+
+import static org.openhab.binding.nikohomecontrol.NikoHomeControlBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.InetAddress;
+import java.util.HashMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.nikohomecontrol.internal.discovery.ThingDiscoveryService;
+import org.openhab.binding.nikohomecontrol.internal.protocol.NikoHomeControlCommunication;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link NikoHomeControlBridgeHandler} is the handler for a Niko Home Control Gateway and connects it to
+ * the framework.
+ *
+ * @author Mark Herwege
+ */
+public class NikoHomeControlBridgeHandler extends BaseBridgeHandler {
+
+    private Logger logger = LoggerFactory.getLogger(NikoHomeControlBridgeHandler.class);
+
+    private NikoHomeControlCommunication nhcComm = null;
+
+    private ScheduledFuture<?> refreshTimer;
+
+    public NikoHomeControlBridgeHandler(Bridge nikoHomeControlBridge) {
+        super(nikoHomeControlBridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // There is nothing to handle in the bridge handler
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.debug("Niko Home Control: initializing bridge handler");
+        InetAddress addr = null;
+        int port;
+
+        Configuration config = this.getConfig();
+        // workaround for int port not being read back as int from configuration, remove trailing .0
+        logger.debug("Niko Home Control: bridge handler port {}", config.get(CONFIG_PORT).toString());
+        String portString = config.get(CONFIG_PORT).toString();
```

Comment:
I saw other people having the same problem, pity, however since they both implement the Number interface this should also work:

```java
int port = ((Number) config.get(CONFIG_PORT)).intValue(); 
```